[Test] Add FakeTensor test for ConvTranspose2d zero-size output

dumko2001 opened the pull request:
Adds a regression test to 

test_fake_tensor.py
 verify zero-size output behavior for ConvTranspose2d in FakeTensorMode, as requested in [review](https://github.com/pytorch/pytorch/pull/171127).

The actual fix was already merged into main, so this PR just adds the missing test coverage to prevent regression.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

zou3519 commented on torch/_meta_registrations.py:
```diff
@@ -2420,7 +2420,7 @@ def _formula_transposed(ln: int, p: int, d: int, k: int, s: int, op: int) -> int
 
     torch._check(
-        sym_or(*[x > 0 for x in ret_shape[2:]]),
+        sym_or(*[x >= 0 for x in ret_shape[2:]]),
```
Can you add a test to test_fake_tensor.py instead? This sounds like a FakeTensor problem, not an Inductor problem

dumko2001 committed:
Add test for ConvTranspose2d zero-size output (fixes #171093)

dumko2001 commented:
@zou3519 just pinging u since the tests have all passed

dumko2001 committed:
Fix test_conv_transpose_zero_size_output for ROCm and PropagateRealTensors

- Skip test on ROCm (HIP) as MIOpen does not support zero-sized outputs.
- Initialize ConvTranspose2d outside FakeTensorMode to support PropagateRealTensors mode.
